unix/vncserver/vncsession.c

```diff
@@ -360,6 +364,25 @@ redir_stdio(const char *homedir, const char *display)
             syslog(LOG_CRIT, "Failure creating \"%s\": %s", logfile, strerror(errno));
             _exit(EX_OSERR);
         }
+
+        result = getfilecon(logfile, &context);
+
+        if (result < 0) {
+            syslog(LOG_CRIT, "Failure obtaining SELinux context for \"%s\"", logfile);
+            _exit(EX_OSERR);
+        }
+
+        if (strstr(context, "vnc_home_t") == NULL) {
```

Comment:
It's always dangerous to hard code a redundancy like this. Can we instead as selinux if the file is "wrong"?

Or if we hard code something, wouldn't it be safer to hard code the known wrong ones?

Reply:
Changed to:
` if (strstr(context, "xdm_home_t") || strstr(context, "user_home_t"))`

I think those should be the wrong ones, but I'll wait for @zpytela.

Reply:
I'd also use functions from libselinux, equivalents of restorecon binary. Entries in `unix/vncserver/selinux/vncsession.fc` read:
```
HOME_DIR/\.vnc(/.*)?      gen_context(system_u:object_r:vnc_home_t,s0)
HOME_ROOT/\.vnc(/.*)?      gen_context(system_u:object_r:vnc_home_t,s0)
```
so these are the correct ones for `~/.vnc/`.

Reply:
@vmojzis I found selinux_file_context_verify() function. Can this be used instead of getting the context of that folder and comparing strings?

Reply:
> @vmojzis I found selinux_file_context_verify() function. Can this be used instead of getting the context of that folder and comparing strings?

Yes, selinux_file_context_verify should be ideal for this situation (it makes sure given file has appropriate context based on the loaded policy). If you still wanted to verify that the context is  "vnc_home_t" you could use getfilecon (gives you the context assigned to given file on disk) or selabel_lookup (gives you the context suggested by the system policy). But I would go with selinux_file_context_verify to avoid hard-coding anything.